### PR TITLE
Changed validator for NewCronjobDialog

### PR DESCRIPTION
**Pull request by drpayyne**
**Description**

Changed validator for `com.magento.idea.magento2plugin.actions.generation.dialog.NewCronjobDialog`

**Fixed Issues**

1. magento/magento2-phpstorm-plugin#299: Change validation for the injection of new cron job dialog window

**Additional Changes**

- Added a new rule `BoxNotEmptyRule` to validate if an option from the dropdown is selected. This can be ported to validate module list dropdown and theme list dropdown as well, while implementing filtering. So we end up getting a validation of 'an option from the pre-populated list is selected' _and_ a filtering the list with an editable field.

**Contribution checklist**
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [x] All new or changed code is covered with integration/functional tests (if applicable)
 - [x] All automated tests passed successfully (all builds are green)

### CODE REVIEW

File: src/com/magento/idea/magento2plugin/actions/generation/dialog/validator/NewCronjobValidator.java [hunk 1]
```diff
@@ -1,285 +0,0 @@
...
-
-            JOptionPane.showMessageDialog(
-                    null,
-                    errorMessage,
-                    errorTitle,
-                    JOptionPane.ERROR_MESSAGE
-            );
-
-            return false;
-        }
```
Comment (drpayyne) [inline src/com/magento/idea/magento2plugin/actions/generation/dialog/validator/NewCronjobValidator.java:281]: This is yet to be implemented. Similar to checking if PHP class exists already.

File: src/com/magento/idea/magento2plugin/actions/generation/dialog/validator/rule/CronScheduleRule.java [hunk 2]
```diff
@@ -0,0 +1,17 @@
+package com.magento.idea.magento2plugin.actions.generation.dialog.validator.rule;
```
Comment (VitaliyBoyko) [inline src/com/magento/idea/magento2plugin/actions/generation/dialog/validator/rule/CronScheduleRule.java:1]: Add copyrights here and everywhere please

**Review by VitaliyBoyko — CHANGES_REQUESTED** [review]

File: src/com/magento/idea/magento2plugin/actions/generation/dialog/validator/rule/BoxNotEmptyRule.java [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+
+    @Override
+    public boolean check(final String value) {
+        return !value.isEmpty();
+    }
+
+    public static ValidationRule getInstance() {
+        return INSTANCE;
+    }
+}
```
Comment (VitaliyBoyko) [inline src/com/magento/idea/magento2plugin/actions/generation/dialog/validator/rule/BoxNotEmptyRule.java:15]: Add line pls

**Review by VitaliyBoyko — APPROVED** [review]
👍

**Pull request merged**